ingestion/src/metadata/ingestion/lineage/parser.py

```diff
@@ -37,214 +43,283 @@
+        """
+        For each table involved in the query, find its joins against any
+        other table.
+        :return: for each table name, list all joins against other tables
+        """
+        join_data = defaultdict(list)
+        # These are @lazy_property, not properly being picked up by IDEs. Ignore the warning
+        for statement in self.parser.statements_parsed:
+            self.stateful_add_joins_from_statement(join_data, statement=statement)
+
+        return join_data
+
+    def retrieve_tables(self, tables: List[Any]) -> List[Table]:
+        if not self._clean_query:
+            return []
+        return [table for table in tables if isinstance(table, Table)]
+
+    @classmethod
+    def clean_raw_query(cls, raw_query: str) -> Optional[str]:
+        """
+        Given a raw query from any input (e.g., view definition,
+        query from logs, etc.), perform a cleaning step
+        before passing it to the LineageRunner
+        """
+        clean_query = insensitive_replace(
+            raw_str=raw_query,
+            to_replace=" copy grants ",  # snowflake specific
+            replace_by=" ",  # remove it as it does not add any value to lineage
         )
 
-        if not table_left or not table_right:
-            logger.warning(f"Cannot find ingredients from {comparison}")
-            continue
-
-        left_table_column = TableColumn(table=table_left, column=column_left)
-        right_table_column = TableColumn(table=table_right, column=column_right)
-
-        # We just send the info once, from Left -> Right.
-        # The backend will prepare the symmetric information.
-        stateful_add_table_joins(join_data, left_table_column, right_table_column)
-
-
-def get_table_joins(
-    parser: LineageRunner, tables: List[str], aliases: Dict[str, str]
-) -> Dict[str, List[TableColumnJoin]]:
-    """
-    For each table involved in the query, find its joins against any
-    other table.
-    :param parser: LineageRunner parser
-    :param tables: involved tables in the query
-    :param aliases: table aliases dict
-    :return: for each table name, list all joins against other tables
-    """
-    join_data = defaultdict(list)
-    for statement in parser.statements_parsed:
-        stateful_add_joins_from_statement(
-            join_data, statement=statement, tables=tables, aliases=aliases
+        clean_query = insensitive_replace(
+            raw_str=clean_query.strip(),
+            to_replace="\n",  # remove line breaks
+            replace_by=" ",
         )
 
-    return join_data
+        if insensitive_match(clean_query, ".*merge into .*when matched.*"):
+            clean_query = insensitive_replace(
+                raw_str=clean_query,
+                to_replace="when matched.*",  # merge into queries specific
+                replace_by="",  # remove it as LineageRunner is not able to perform the lineage
+            )
+
+        if insensitive_match(clean_query, "^COPY.*"):
```

Comment:
is this the one for https://github.com/open-metadata/OpenMetadata/issues/6203? Could we add some info around the why in the comments itself pls?

Reply:
Comment added.